redash/tasks/queries/execution.py

```diff
@@ -224,12 +222,10 @@ def run(self):
                 utcnow(),
             )
 
-            updated_query_ids = models.Query.update_latest_result(query_result)
```

Comment:
I missed this update... Could you let me know why you removed this line?

Reply:
QueryResult and Query have relationship, there's no need to run multiple queries to store both separately (especially, when Query object already exists), doing everything in `store_result` function

Reply:
It looks like that Query is  updated in `update_latest_result`. 
https://github.com/getredash/redash/blob/b7f22b1896454b495f4a0412cf6beb11be78b466/redash/models/__init__.py#L741

Can we skip this update?

Reply:
update_latest_result method was removed

Reply:
Query was updated in the method(as below). 

https://github.com/getredash/redash/pull/6671/files#diff-8b1469306b4cd26cae17218f41c75c810d981099b0b11fa48aec8101c7a8c741L737-L741

Is there an alternative to this? Or have you decided that this process is not necessary? If you decide that it is unnecessary, please let us know the reason.

It seems to me that without this update, Redash's behavior will be affected. Isn't Query.lastest_query_data and some attributes no longer updated?

Reply:
@masayuki038 removed update_latest_result, cause all changes are happening in QueryResult.store_result

Reply:
Thanks for your explanation. I confirmed you updated the definition of `last_query_data` in Query on this PR.
Why did you name 'queries` in QeuryResult? Query - QueryResult relationship is 1:n, right? Or m:n?

Reply:
one QueryResult record can be the same for multiple queries, which have the same hash and the same datasource -> 1:n. At the same time one query can have multiple query results, but redash API relies only on the latest result and to query historical data it's possible to use separate QueryResult data source

Reply:
> 1:n - one QueryResult record can be the same for multiple queries with same query hash and datasource id

Does the "1:n" mean one QueryReulst / multiple Queries, right?

Reply:
exactly

Reply:
I didn't add `queries` relationship, it existed before. Adding it is a part of [migration](https://docs.sqlalchemy.org/en/20/orm/relationship_api.html#sqlalchemy.orm.relationship.params.backref) from legacy `backref` argument to explicit relationships with `back_populates`

Reply:
It seems to be in this PR... Am I missing anything?
![image](https://github.com/getredash/redash/assets/36331/d203621b-6997-4cf1-8088-e336464619fe)

Reply:
with migration from `backref` to `back_populates` `queries` relationship should be set explicitly, with `backref` it existed implicitly

Reply:
> queries relationship should be set explicitly

OK. So, you defined `queries`, right?

Reply:
right, it was a part of migration

Reply:
OK. The schema definition of `query_results` is here:

```
postgres-# \d query_results;
                                         Table "public.query_results"
     Column     |           Type           | Collation | Nullable |                  Default
----------------+--------------------------+-----------+----------+-------------------------------------------
 id             | integer                  |           | not null | nextval('query_results_id_seq'::regclass)
 org_id         | integer                  |           | not null |
 data_source_id | integer                  |           | not null |
 query_hash     | character varying(32)    |           | not null |
 query          | text                     |           | not null |
 data           | text                     |           |          |
 runtime        | double precision         |           | not null |
 retrieved_at   | timestamp with time zone |           | not null |
Indexes:
    "query_results_pkey" PRIMARY KEY, btree (id)
    "ix_query_results_query_hash" btree (query_hash)
Foreign-key constraints:
    "query_results_data_source_id_fkey" FOREIGN KEY (data_source_id) REFERENCES data_sources(id)
    "query_results_org_id_fkey" FOREIGN KEY (org_id) REFERENCES organizations(id)
Referenced by:
    TABLE "queries" CONSTRAINT "queries_latest_query_data_id_fkey" FOREIGN KEY (latest_query_data_id) REFERENCES query_results(id)
```

It looks like that one Query / multiple QueryResults.
Why did you define and name `queries` in QueryResult?

Reply:
@masayuki038 please check implementation of removed update_latest_result method. It retrieves all queries by hash and datasource id and updates them in a loop with a latest query result.

Reply:
> please check implementation of removed update_latest_result method. It retrieves all queries by hash and datasource id and updates them in a loop with a latest query result.

Yes, I understand that. I don't think there is any problem since `update_latest_result ` doesn't use relationship, 

However, in this update, it is defined as a relationship to set Query.`last_query_data`. And I'm concerned that the content is defined differently from the relationship in the DB schema. It looks like there is an inconsistency in the relationship between DB schema (Query: QueryResult = 1:N) and SQLAlchemy (Query: QueryResult = N:M?).

Is it reasonable to define such a conflicting relationship in order to update the Query's `last_query_data` when creating a QueryResult?

Reply:
i do not see contradiction, from what you've posted. 
```
Referenced by:
   TABLE "queries" CONSTRAINT "queries_latest_query_data_id_fkey" FOREIGN KEY (latest_query_data_id) REFERENCES query_results(id)
```
so 1 QueryResult to many Queries

Reply:
I don't think that having Foreign Key constraints has anything to do with multiplicity.

I thought 1 Query to many QueryResults (like this).
![image](https://github.com/getredash/redash/assets/36331/e6f6c6ce-a337-48df-b237-25f8f137e003)

Is it not correct?

Reply:
Query has latest_query_data_id attribute, which references private key in QueryResult table, which means, that multiple Query object can set the same latest_query_data_id, which means that 1 QueryResult will be referred by multiple Queries - 1 Query - many QueryResults

Reply:
hey @masayuki038 
sorry for time you've spend in a review process, but closing this PR as not going to work on redash any further

Reply:
> which means, that multiple Query object can set the same latest_query_data_id,

This is the case if different Queries have the same `query_hash`. However, it is caused by hash value collisions, right? I don't think we should discuss it in this context.

> 1 Query - many QueryResults

Yes. Therefore, I think we should name `query` instead of `queries` in QueryResult. That is my point.